exploration/text-vs-code.md

```diff
@@ -0,0 +1,216 @@
+
+As MessageFormat 2 will be at best a secondary language to all its users,
+it should conform to user expectations and require as little learning as possible.
+
+The syntax should avoid footguns,
+in particular as it's passed through various tools during formatting.
+
+ASCII-compatible syntax. While support for non-ASCII characters for variable names,
+values, literals, options, and the like are important, the syntax itself should
+be restricted to ASCII characters. This allows the message to be parsed
+visually by humans even when embedded in a syntax that requires escaping.
+
+Whitespace is forgiving. We _require_ the minimum amount of whitespace and allow
+users to format or change unimportant whitespace as much as they want.
+This avoids the need for translators or tools to be super pedantic about
+formatting.
+
+## Constraints
+
+Limiting the range of characters that need to be escaped in plain text is important.
+Following past precedent,
+this design doc will only consider encapsulation styles which
+start with `{` and end with `}`.
+
+The current syntax includes some plain-ascii keywords:
+`input`, `local`, `match`, and `when`.
+
+The current syntax and active proposals include some sigil + name combinations,
+such as `:number`, `$var`, `|literal|`, `+bold`, and `@attr`.
+
+The current syntax supports unquoted literal values as operands.
+
+Messages themselves are "simple strings" and must be considered to be a single
+line of text. In many containing formats, newlines will be represented as the local
+equivalent of `\n`.
+
+## Proposed Design
+
+### Start in text, encapsulate code, trim around statements
+
+Allow for message patterns to not be quoted.
+
+Encapsulate with `{…}` or otherwise distinguishing statements from
+the primarily unquoted translatable message contents.
+
+For messages with multiple variants,
+separate the variants using `when` statements.
+
+Trim whitespace between and around statements such as `input` and `when`,
+but do not otherwise trim any leading or trailing whitespace from a message.
+This allows for whitespace such as spaces and newlines to be used outside patterns
+to make a message more readable.
+
+Allow for a pattern to be `{{…}}` quoted
+such that it preserves its leading and/or trailing whitespace
+even when preceded or followed by statements.
+
+## Alternatives Considered
+
+### Start in code, encapsulate text
+
+This approach treats messages as something like a resource format for pattern values.
+Keywords are declared directly at the top level of a message,
+and patterns are always surrounded by `{{…}}` or some other delimiters.
+
+Whitespace in patterns is never trimmed.
+
+The `{{…}}` are required for all messages,
+including ones that only consist of text.
+Delimiters of the resource format are required in addition to this,
+so messages may appear wrapped as e.g. `"{{…}}"`.
+
+This option is not chosen due to adding an excessive
+quoting burden on all messages.
```

Comment:
I think we should not include these "This option is not chosen due..." paragraphs. I think it is okay to call out objective or subjective reasons for why we might not choose a given alternative, e.g.

```suggestion
- This option makes plain text strings invalid as messages.
- This option requires additional quoting for simple messages.
```

Our choice section should deal with the logic of why a given option was chosen.

Reply:
Could you clarify what you mean by "choice section"? I'm not sure that I understand what that is.